returnn/tf/layers/base.py

```diff
@@ -1380,7 +1380,7 @@ def get_constraints_value(self):
       c += self.spatial_smoothing * self.get_output_spatial_smoothing_energy()
     if self.darc1:
       c += self.darc1 * self.get_darc1()
-    if c is 0:
+    if c == 0:
```

Comment:
This is wrong in case `c` is a `tf.Tensor` or `numpy.ndarray`.

Same also for all the other changes.